src/modules/Dropdown/Dropdown.js

```diff
@@ -227,6 +246,25 @@ export default class Dropdown extends Component {
       debug('value changed, setting', nextProps.value)
       this.setValue(nextProps.value)
     }
+
+    if (!_.isEqual(nextProps.options, this.props.options)) {
+      if (_.first(this.state.options) !== _.first(this.props.options)) {
+        // `Add: ...` option at the top of the list
+        this.setState({
+          options: [_.first(this.state.options), ...nextProps.options],
+        })
+      } else if (_.last(this.state.options) !== _.last(this.props.options)) {
+        // `Add: ...` option at the bottom of the list
+        this.setState({
+          options: [...nextProps.options, _.last(this.state.options)],
+        })
+      } else {
+        // No `Add: ...` option -> just copy the list
+        this.setState({
+          options: [...nextProps.options],
+        })
```

Comment:
Instead of putting `options` on state, I think we should insert the `Add:` item into the array of items returned by `getMenuOptions()` method.  This should be the only place it is required.  Anytime the menu renderers, it will have the `Add:` item in it if necessary.

This would allow us to remove the state and extra logic from many places. The current code for handling the `onAddition` callback will still work.

```js
getMenuOptions = () => {
  // ... at the bottom of the function:
  
  // insert the "add" item
  if (allowAdditions && search && searchQuery && !_.some(filteredOptions, { text: searchQuery })) {
    const addItem = {
      text: `Add: ${searchQuery}`,
      value: searchQuery,
    }
    if (additionPosition === 'top') filteredOptions.unshift(addItem)
    else filteredOptions.push(addItem)
  }

  return filteredOptions
}
```

We can then save a render and we can remove all the code that is handling `options` on state.